components/director/internal/runtimemapping/token_verifier.go

```diff
@@ -98,10 +98,14 @@ func (f *jwksFetch) GetKey(ctx context.Context, token *jwt.Token) (interface{},
 		return nil, errors.Wrap(err, "while getting the key ID")
 	}
 
-	keys := jwksSet.LookupKeyID(keyID)
-	for _, key := range keys {
-		if key.Algorithm() == token.Method.Alg() {
-			return key.Materialize()
+	if keys, isFound := jwksSet.LookupKeyID(keyID); isFound {
```

Comment:
> * `(jwk.Set).LookupKeyID()` no longer returns an array of `jwk.Key`.
    Instead, only the first key matching the given key ID will be returned.
    If you need to work with multiple keys, use `(jwk.Set).Iterate()` or
    `(jwk.Set).Get()` to look for matching keys.

Previously all the keys matching the ID were returned and we iterated them to find the one with the right algorithm.

Right now only the first is returned and if it is not with the right Alg then we miss it.

You should use Iterate/Walk again here.